Buildings/Applications/DHC/CentralPlants/Cooling/Controls/ChillerStage.mo

```diff
@@ -0,0 +1,152 @@
+  Modelica.Blocks.Interfaces.RealOutput y[2]
+    "On/off signal for the chillers - 0: off; 1: on"
+    annotation (Placement(transformation(extent={{100,-10},{120,10}})));
+  Modelica.StateGraph.InitialStep off(nIn=1) "No cooling is demanded"
+    annotation (Placement(transformation(
+        extent={{-10,10},{10,-10}},
+        rotation=-90,
+        origin={-50,70})));
+  Modelica.StateGraph.StepWithSignal oneOn(nOut=2, nIn=2)
+    "Status of one chiller on"
+    annotation (Placement(
+        transformation(
+        extent={{10,-10},{-10,10}},
+        rotation=90,
+        origin={-50,0})));
+  Modelica.StateGraph.StepWithSignal twoOn
+    "Status of two chillers on"
+    annotation (Placement(
+        transformation(
+        extent={{10,-10},{-10,10}},
+        rotation=90,
+        origin={-50,-70})));
+  Modelica.StateGraph.Transition offToOne(
+    condition=on == true,
+    enableTimer=true,
+    waitTime=tWai) "Condition of transition from off to one chiller on"
+    annotation (Placement(transformation(
+        extent={{10,10},{-10,-10}},
+        rotation=90,
+        origin={-50,40})));
+  Modelica.StateGraph.Transition oneToTwo(
+    enableTimer=true,
+    waitTime=tWai,
+    condition=-QLoa >= -(criPoiLoa + dQ))
+    "Condition of transition from one chiller to two chillers"
+    annotation (
+      Placement(transformation(
+        extent={{10,10},{-10,-10}},
+        rotation=90,
+        origin={-50,-40})));
+  Modelica.StateGraph.Transition twoToOne(
+    enableTimer=true,
+    waitTime=tWai,
+    condition=-QLoa < -(criPoiLoa - dQ))
+    "Condition of transion from two chillers to one chiller"
+    annotation (
+      Placement(transformation(
+        extent={{-10,10},{10,-10}},
+        rotation=90,
+        origin={0,-40})));
+  Modelica.StateGraph.Transition oneToOff(
+    condition=on == false,
+    enableTimer=true,
+    waitTime=tWai) "Condition of transition from one chiller to off"
+    annotation (Placement(
+        transformation(
+        extent={{-10,10},{10,-10}},
+        rotation=90,
+        origin={-20,40})));
+  inner Modelica.StateGraph.StateGraphRoot stateGraphRoot
+    "State graph root"
+    annotation (Placement(transformation(extent={{60,60},{80,80}})));
+  Modelica.Blocks.Tables.CombiTable1Ds combiTable1Ds(table=[0,0,0; 1,1,0; 2,1,1])
+    "Table for interpretation of chiller on/off status output"
+    annotation (Placement(transformation(extent={{70,-10},{90,10}})));
+  Buildings.Controls.OBC.CDL.Conversions.BooleanToInteger booToInt(
+    final integerTrue=1, final integerFalse=0)
+    "Boolean to real"
+    annotation (Placement(transformation(extent={{30,-40},{50,-20}})));
+  Buildings.Controls.OBC.CDL.Conversions.BooleanToInteger booToInt1(
+    final integerTrue=2, final integerFalse=0)
+    "Boolean to real"
+    annotation (Placement(transformation(extent={{30,-80},{50,-60}})));
+  Buildings.Controls.OBC.CDL.Integers.Add addInt
+    "Adding up integer signals"
```

Comment:
is this block basically calculating the chiller stage index, say stage 0, 1, 2? If yes, then you may want to update the comment.